### PR TITLE
[pro#279] Dashboard batch request CSV export

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Fixes https://github.com/mysociety/alaveteli-professional/issues/279
Fixes https://github.com/mysociety/transparency-adessium/issues/9

## What does this do?

Adds links export links to info request batches on the dashboard. Links will generate and download a CSV with metrics for the batch. EG. request url, authority name, number of replies & request status.

## Screenshots

image

### CODE REVIEW

File: app/services/info_request_batch_metrics.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+##
+#
+#
```
Comment (garethrees) [inline app/services/info_request_batch_metrics.rb]: Should include documentation

File: app/controllers/alaveteli_pro/downloads_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+##
+# Controller which managed Alaveteli Professional request/batch data downloads.
+#
+class AlaveteliPro::DownloadsController < AlaveteliPro::BaseController
+  def show
+    authorize! :download, info_request_batch
+
+    respond_to do |format|
+      format.html { head :bad_request }
+      format.csv do
```
[the remark this answers is not in the document]
Comment (gbp) [reply app/controllers/alaveteli_pro/downloads_controller.rb]: Yep exactly.

File: app/controllers/alaveteli_pro/batch_downloads_controller.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ##
-# Controller which managed Alaveteli Professional request/batch data downloads.
+# Controller which managed Alaveteli Professional info request batch data
```
Comment (garethrees) [inline app/controllers/alaveteli_pro/batch_downloads_controller.rb]: s/managed/manages/

File: app/services/info_request_batch_metrics.rb [hunk 4]
```diff
@@ -23,22 +20,22 @@ def metrics
 
       { request_url: url,
         authority_name: info_request.public_body.name,
-        number_of_replies: info_request.incoming_messages.size,
+        number_of_replies: info_request.incoming_messages_count,
```
Comment (garethrees) [inline app/services/info_request_batch_metrics.rb]: Calling `#size` is fine as that defaults to the counter-cache value:

```ruby
r.incoming_messages.size
# => 0
r.incoming_messages.count
# (13.1ms)  SELECT COUNT(*) FROM "incoming_messages" WHERE "incoming_messages"."info_request_id" = $1  [["info_request_id", 114]]
# => 0
```

**Pull request merged**